data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetDataColumnSidecars.java

```diff
@@ -0,0 +1,110 @@
+
+  public GetDataColumnSidecars(
+      final DataProvider dataProvider, final SchemaDefinitionCache schemaCache) {
+    this(dataProvider.getChainDataProvider(), schemaCache);
+  }
+
+  public GetDataColumnSidecars(
+      final ChainDataProvider chainDataProvider, final SchemaDefinitionCache schemaCache) {
+    super(createEndpointMetadata(schemaCache));
+    this.chainDataProvider = chainDataProvider;
+  }
+
+  private static EndpointMetadata createEndpointMetadata(final SchemaDefinitionCache schemaCache) {
+    return EndpointMetadata.get(ROUTE)
+        .operationId("getDataColumnSidecars")
+        .summary("Get data column sidecars")
+        .description(
+            "Retrieves data column sidecars for a given block id.\n"
+                + "    Depending on `Accept` header it can be returned either as json or as bytes serialized by SSZ.\n"
+                + "    If the `indices` parameter is specified, only the data column sidecars with the specified indices will be returned. There are no guarantees\n"
+                + "    for the returned data column sidecars in terms of ordering.")
+        .tags(TAG_BEACON)
+        .pathParam(PARAMETER_BLOCK_ID)
+        .queryListParam(DATA_COLUMN_INDICES_PARAMETER)
+        .response(SC_OK, "Request successful", getResponseType(schemaCache), getSszResponseType())
+        .withNotFoundResponse()
+        .build();
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final List<UInt64> indices = request.getQueryParameterList(DATA_COLUMN_INDICES_PARAMETER);
+    final SafeFuture<Optional<List<DataColumnSidecar>>> future =
+        chainDataProvider.getDataColumnSidecars(
+            request.getPathParameter(PARAMETER_BLOCK_ID), indices);
+
+    request.respondAsync(
+        future.thenApply(
+            dataColumnSidecars ->
+                dataColumnSidecars
+                    .map(AsyncApiResponse::respondOk)
+                    .orElse(AsyncApiResponse.respondNotFound())));
+  }
+
+  private static SerializableTypeDefinition<List<DataColumnSidecar>> getResponseType(
+      final SchemaDefinitionCache schemaCache) {
+    final DeserializableTypeDefinition<DataColumnSidecar> dataColumnSidecarType =
+        SchemaDefinitionsEip7594.required(schemaCache.getSchemaDefinition(SpecMilestone.EIP7594))
+            .getDataColumnSidecarSchema()
+            .getJsonTypeDefinition();
+    return SerializableTypeDefinition.<List<DataColumnSidecar>>object()
+        .name("GetDataColumnSidecarsResponse")
+        .withField("data", listOf(dataColumnSidecarType), Function.identity())
+        .build();
+  }
+
+  private static ResponseContentTypeDefinition<List<DataColumnSidecar>> getSszResponseType() {
+    final OctetStreamResponseContentTypeDefinition.OctetStreamSerializer<List<DataColumnSidecar>>
+        serializer =
+            (data, out) -> data.forEach(dataColumnSidecar -> dataColumnSidecar.sszSerialize(out));
```

Comment:
I'm not sure if I'm not missing anything, but how should we parse a list of SSZ objects glued together in the returned stream. Shouldn't we serialize the whole list as `SszList<DataColumnSidecar>` ssz type?